draft5/app/controllers/ValidatorController.js

```diff
@@ -133,4 +144,17 @@ app.controller('validatorController', function ($scope, $http, $window) {
     self.validateDocument();
   });
 
+  // Save form data before reload
+  $window['onbeforeunload'] = function () {
```

Comment:
I don't think so? I'm not actually super familiar with AngularJS, so I'm not 100% sure. Googling shows the most common solution for adding a function that cleans up before a page closes/reloads is using onbeforeunload. If I'm not mistaken, there isn't an event I can listen for in AngularJS that would give me the same effect, because hitting refresh or closing the tab/window happens instantly and it interrupts whatever Angular is doing, unless you've defined onbeforeunload. 

That is, at least, my understanding of the problem. It is possible that I'm just not googling the right questions.